python/lsst/queued_butler_writer/butler.py

```diff
@@ -91,7 +92,8 @@ def _insert_dimension_records(butler: Butler, records: list[DimensionRecord]) ->
     dimensions = butler.dimensions.sorted(grouped_records.keys())
     for dimension in dimensions:
         records = grouped_records[dimension.name]
-        butler.registry.insertDimensionData(dimension, *records, skip_existing=True)
+        with time_this(_LOG, msg=f"inserted {len(records)} dimension data records", level=logging.DEBUG):
+            butler.registry.insertDimensionData(dimension, *records, skip_existing=True)
```

Comment:
I suggest moving the time block to outside the dimensions loop. Apologies for not catching this before testing.